decimalSuite_test.go

```diff
@@ -50,23 +55,34 @@ func TestFromSuite(t *testing.T) {
 			if PrintFiles {
 				fmt.Println("starting test:", file)
 			}
-			dat, _ := ioutil.ReadFile(file)
-			dataString := string(dat)
-			testVals := getInput(dataString)
-			numTests := len(testVals)
+			f, _ := os.Open(file)
+			scanner := bufio.NewScanner(f)
+			numTests := 0
 			failedTests := 0
-			for _, testVal := range testVals {
-				dec64vals := convertToDec64(testVal)
-				calcRestul, testErr := runTest(dec64vals, testVal)
-				if PrintTests {
-					fmt.Printf("%s %s %v %v %v -> %v\n", testVal.testName, testVal.testFunc, testVal.val1, testVal.val2, testVal.val3, testVal.expectedResult)
+			var roundingSupported bool
+			var scannedContext Context64
+
+			for scanner.Scan() {
+				testVal, err := getInput(scanner.Text())
+				if testVal.rounding != "" {
```

Comment:
If `err` is `nil`, can you rely on the value of `testVal`? This is normally considered an anti-pattern.

Reply:
yes
function no longer returns an error